spond/spond.py

```diff
@@ -308,13 +314,19 @@ async def get_event(self, uid) -> dict:
         Returns
         -------
         Details of the event.
+
+        Raises
+        ------
+        IndexError if no event is matched.
+
         """
         if not self.events:
             await self.get_events()
         for event in self.events:
             if event["id"] == uid:
                 return event
-        raise IndexError
+        errmsg = f"No event with id='{uid}'"
+        raise IndexError(errmsg)
```

Comment:
A more common pattern is:

```
if not self.events:
    raise ValueError(f"No event with id='{uid}'")
for event in self.events:
...
```
`ValueError` feels more appropriate than `IndexError` but that's just my take based on the [docs](https://docs.python.org/3/library/exceptions.html#ValueError)

Reply:
But this is not the correct check.
`self.events` can contain lots of events.
What we are looking for is a _specific_ event.

Now, we could probably improve this with a lambda-function or something like
```
return next(filter(lambda x: x['id'] == uid, self.events))
```
But I am not sure it will actually be any faster or better, or if it improves readbility.

Anyway we are raising in the end if the event we are looking for does not exist.

Reply:
Oh sorry I had misread the code, you're right, my suggestion is incorrect. What about 

```
return next(e for e in self.events if e["id"] == uid)
```

I think this expression will return `None` if there is no matching event based on uid

Reply:
But I am not sure we want to return None, as that requires a check in the calling code instead. Now we raise an exception already here if no event is found.

Reply:
Yeah agreed, probably not the best solution, we can do a check after to raise an error

```
found_event = next(e for e in self.events if e["id"] == uid)
if not found_event:
    raise ValueError(f"Event with {uid} not found")
```

Reply:
Looks like I resolved this without committing my comments x 2 ... 🤦🤦

- I think the original pattern is adequate; using `next` would require handling a `StopIteration` exception here, which seems to add unecessary complexity.

- I've kept the type of exception raised as-is; but will open a separate issue for discussion as I have some thoughts that neither `IndexError` nor `ValueError` are ideal.

Reply:
I agree that the next() solution is not actually fixing anything. So a loop is just as good